### PR TITLE
[docs] remove walkthrough page and references

### PR DESCRIPTION
**Pull request by kbrandwijk**
# Why

The walkthrough page is outdated. We are removing it until it is replaced.

# How

Removed page from navigation, added redirects in both places, updated incoming links to it.

# Test Plan

Tested locally for navigation and redirects.

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/bare/exploring-bare-workflow.md [hunk 1]
```diff
@@ -69,4 +69,4 @@ With [Expo Application Services (EAS)](/eas/index.md), you can build and submit
 
 You are now, at a very high level, familiar with the steps you would go through to get started on building an app with the bare workflow. Continue on to [Up and Running](hello-world.md) to get started coding!
 
-Are you feeling intimidated? It might be better for you to start out with the managed workflow if you're new to this. Check out the [managed workflow walkthrough](../introduction/walkthrough.md) for more information.
+Are you feeling intimidated? It might be better for you to start out with the managed workflow if you're new to this. Check out the [First steps](../tutorial/planning/) for more information.
```
Comment (brentvatne) [inline docs/pages/bare/exploring-bare-workflow.md:72]: maybe we should remove this exploring bare workflow page too? it's the same sort of walkthrough

Comment (kbrandwijk) [reply docs/pages/bare/exploring-bare-workflow.md:72]: I updated it to be sure, but I think the page is already removed actually.

Comment (brentvatne) [reply docs/pages/bare/exploring-bare-workflow.md:72]: it still lives here https://docs.expo.dev/bare/exploring-bare-workflow/

**Review by brentvatne — APPROVED** [review]
looks good although i think we can remove https://docs.expo.dev/bare/exploring-bare-workflow/

**Pull request merged**